### PR TITLE
Add download module

### PR DESCRIPTION
**Pull request by roshankern**
@gwaybio Sorry forgot to tag you! This PR is ready for review.

I think this download module is mostly complete content wise (except for your suggestions of course). I could also imagine adding content once we design a streaming pipeline for processing the entire dataset that is >20 TB. 

I did have some questions about the `environment.yaml` file. I'm sure mine is completely wrong, I just used `conda env export` command to create it. I assume I should follow the conventions shown in the [cell-health repo](https://github.com/broadinstitute/cell-health/blob/master/environment.yml). Regarding this file:
- Why use conda-forge channel instead of default? Should I always be using conda-forge channel?
- What packages are necessary to include in the `environment.yaml` file? Can I include only the things I had to do `conda install` for? Conda will automatically install dependencies, correct?

### CODE REVIEW

**Comment by roshankern** [comment]
@gwaybio Sorry forgot to tag you!

File: 0.download_data/0.download_training_movies.sh [hunk 1]
```diff
@@ -0,0 +1,2 @@
+jupyter nbconvert --to python aspera_training_downloader.ipynb
```
Comment (gwaybio) [inline 0.download_data/0.download_training_movies.sh:2]:
```suggestion
#!/bin/bash
jupyter nbconvert --to python aspera_training_downloader.ipynb
```

You need a bash shebang (`#!`) right?

File: 0.download_data/0.download_training_movies.sh [hunk 2]
```diff
@@ -0,0 +1,2 @@
...
+python aspera_training_downloader.py
```
Comment (gwaybio) [inline 0.download_data/0.download_training_movies.sh]:
```suggestion
python aspera_training_downloader.py

```

To remove that red symbol at the end of this file (add a new line character)

File: 0.download_data/README.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# Download MitoCheck Mitosis Movies
+
+In this module, we download all timelapse confocal microscopy mitosis movies generated by [MitoCheck](https://www.mitocheck.org/). 
+All of the single cell profiles are publicly available and hosted by Open Microscopy.
```
Comment (gwaybio) [inline 0.download_data/README.md]:
```suggestion
All of the movies are publicly available and hosted by Open Microscopy.
```

Is this more accurate? I don't know what you mean by single cell profiles.

File: 0.download_data/README.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+  Note: `username` is your username and path_to_aspera can be found with `ascli config ascp show`
+  
+4) Save and exit the editor
+
+## Step 2: Download Labeled Data with Aspera
+
+Use [Aspera training downloader](aspera_training_downloader.ipynb) to download the raw data that has been manually annotated by Mitocheck as described in the [original paper](https://www.nature.com/articles/nature08869#Sec2).
+The downloader works in 2 steps:
+1) This downloader first determines which wells from which plates have labels (from training data sent by Heriche) and saves these plates/wells to `training_locations.tsv`.
```
Comment (gwaybio) [inline 0.download_data/README.md]:
```suggestion
1) This downloader first determines which wells from which plates have labels (from training data sent by J-K Heriche) and saves these plates/wells to `training_locations.tsv`.
```

File: 0.download_data/asperaweb_id_dsa.openssh [hunk 5]
```diff
@@ -0,0 +1,12 @@
+-----BEGIN DSA PRIVATE KEY-----
```
Comment (gwaybio) [inline 0.download_data/asperaweb_id_dsa.openssh]: You should never add private keys to a public repo! You should probably recreate this private key, or delete this commit from history :)

File: 0.download_data/aspera_training_downloader.py [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+# In[3]:
+
+
+#Read plates listed in features dataset to figure out which wells from which plates have labeled data
+#Save these training locations into a file
+
+def save_training_wells(features_path, save_path):
+    data_list = []
+    for plate in os.listdir(features_path):
```
Comment (gwaybio) [inline 0.download_data/aspera_training_downloader.py]: Have you used [`pathlib`](https://docs.python.org/3/library/pathlib.html) before? I find it to be much more intuitive and pythonic than `os`. You might want to consider switching to pathlib.

File: 0.download_data/aspera_training_downloader.py [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        for well in os.listdir(features_path + plate + "/features"):
```
Comment (gwaybio) [inline 0.download_data/aspera_training_downloader.py]: Check out [f-strings](https://realpython.com/python-f-strings/)!

```python
        for well in os.listdir(f"{features_path}{plate}/features"):
```

er, well, with pathlib.

```python
        full_feature_path = pathlib.Path(f"{features_path}{plate}/features")
        for well in full_feature_path.iterdir():
```

**Review by gwaybio — CHANGES_REQUESTED** [review]
Before this is merged, you at least need to remove your private key 😬 

I also made a couple suggestions to the python code: you might consider switching to pathlib, and to using f-strings.

**Comment by gwaybio** [comment]
> I did have some questions about the environment.yaml file. I'm sure mine is completely wrong, I just used conda env export command to create it. I assume I should follow the conventions shown in the [cell-health repo](https://github.com/broadinstitute/cell-health/blob/master/environment.yml). Regarding this file:
>
> Why use conda-forge channel instead of default? Should I always be using conda-forge channel?

conda-forge is heavily maintained and easy to contribute to. I don't know if this is true of defaults, but conda-forge has nearly everything that we need (if not on conda-forge, it will be on [bioconda](https://bioconda.github.io/)). If not on either, then we can easily add to either (most likely conda-forge)

> What packages are necessary to include in the environment.yaml file? Can I include only the things I had to do conda install for? Conda will automatically install dependencies, correct?

Yes, only include things that you would conda install. Managing a conda environment is a balancing act between solvability and reproducibility :)

File: 0.download_data/asperaweb_id_dsa.openssh (hunk 5 above)
Comment (roshankern) [reply 0.download_data/asperaweb_id_dsa.openssh]: Yikes! Recreated the key.

File: 0.download_data/aspera_training_downloader.py (hunk 6 above)
Comment (roshankern) [reply 0.download_data/aspera_training_downloader.py]: Done 👍

File: 0.download_data/aspera_training_downloader.py (hunk 7 above)
Comment (roshankern) [reply 0.download_data/aspera_training_downloader.py]: 👍

**Comment by roshankern** [comment]
@gwaybio This should be good to go!

**Review by gwaybio — APPROVED** [review]
nice work!

**Pull request merged**